util.c

```diff
@@ -1253,7 +1253,7 @@ struct supertype *super_by_fd(int fd, char **subarrayp)
 			*subarray++ = '\0';
 			subarray = xstrdup(subarray);
 		}
-		strcpy(container, dev);
+		snprintf(container, sizeof(container) - 1, "%s", dev);
```

Comment:
with `snprintf`,  sizeof(container) - 1 is not needed. You can use  sizeof(container)